### DISCUSSION TITLE
[Ares bug] Type Conversion logic from Ares didn't convert the spawn unit after convert the unit itself.

### DISCUSSION THREAD

**Discussion by unknown-men** [Phobos-developers/Phobos#1087] [General]
Example: When V3 upgrade to V3B unit they still shooting V3ROCKET rather than V3ROCKETB (Preplaced V3B still conrrectly launch V3ROCKETB), My expectations is V3B should launch V3ROCKETB after upgraded like preplaced one.

I test with a lastest nightly build version of Phobos and they didn't fix the bug yet.

**Comment by DOOMGUY2K** [comment]
https://github.com/Phobos-developers/Phobos/pull/1090
